sdk/core/Azure.Core/src/Shared/RetriableStream.cs

```diff
@@ -95,7 +119,10 @@ public override async Task<int> ReadAsync(byte[] buffer, int offset, int count,
                 {
                     try
                     {
-                        var result = await _currentStream.ReadAsync(buffer, offset, count, cancellationToken).ConfigureAwait(false);
+                        int result = await _currentStream.ReadAsync(buffer, offset, count, cancellationToken).ConfigureAwait(false);
+
+                        _onRead?.Invoke(buffer, offset, result);
```

Comment:
Is it expected that `buffer` is writable here?

Reply:
We don't intend to modify it in the _onRead method, good call out.  Is there a way to make it readonly?